prov/rxm/src/rxm_cq.c

```diff
@@ -468,6 +468,13 @@ static ssize_t rxm_rndv_xfer(struct rxm_ep *rxm_ep, struct fid_ep *msg_ep,
 	struct iovec iov[RXM_IOV_LIMIT];
 	void *desc[RXM_IOV_LIMIT];
 	ssize_t ret = FI_SUCCESS;
+	ssize_t conn_sts = FI_SUCCESS;;
+
+	if (rxm_ep->rndv_ops == &rxm_rndv_ops_read) {
+		conn_sts = rxm_check_duplex_conn(&conn->handle);
+		if (conn_sts && conn_sts != -FI_EAGAIN)
+			return conn_sts;
+	}
```

Comment:
> It seems that the check for a usable connection should have occurred sooner than this call.

The intent of delaying the check for the TX connection back to the originator of a request was to simplify the initialization of a deferred entry should the connection in that direction not yet exist. Where the check is now we can easily initialize the defer structure and just let the existing code do that initialization and enqueue it, in addition at this point the RX buffer is no longer needed. Another consideration was to avoid introducing the overhead of the check until it was in code that actually needed to send something back to originator of the software protocol request. With that said, I had debated putting the check earlier in the RX CQ handling processing and it can be done. It would need to be deep enough into the CQ path so that only protocols that require sending something back to the requestor make the check, as otherwise we would create a connection that is not required (i.e. if only RMA and eager processing we may not require a connection back to the requestor depending on application usage). If moved I would want to create a new defer type and hang on to the RX buffer until the deferred entry could be processed, allowing us to re-enter the RX CQ processing. This adds more new code but might make it easier to understand. The new deferred processing would need to first validate the connection (so we don't create duplicate deferred entries for the same request) and then call back into the *right* spot of the CQ processing path if the TX connection has completed. I had taken some comfort in not creating a new code path. Do the extra details alter your opinion or do you want me to investigate moving it up earlier in the CQ processing? Thanks.